src/reports/timeline.rkt

```diff
@@ -244,7 +244,7 @@
                                       (td ,(~a n))))))))
 
 (define (simple-render-phase info name)
-  (if (> (length (first info)) 0)
+  (if (positive? (length (first info)))
```

Comment:
I wish Racket has something like `compare_length_with` in OCaml.

For this particular instance, `(pair? (first info))` is close, but loses the implicit assertion that `(first info)` must satisfy `list?`.

Reply:
Perhaps Resyntax ought to rewrite this to `(not (empty? (first info)))` instead.

Reply:
I think this is a case where keeping falsity so limited really hurts Racket.

Reply:
Not empty is ugly because branching on a not suggests flipping the branches, putting the edge case (empty) first.

Reply:
Also I’ll be honest I don’t find positive? and zero? to be useful for readability.

Reply:
A rule that flips `(if (not foo) long-expr very-short-expr)` into `(if foo very-short-expr long-expr)` might be worth exploring, in combination with one that replaces a length check with an emptiness check. I'm of the opinion that conditionals like that where one branch is a trivial base case and the other has actual logic should be written with the base case first. Plus the emptiness check is constant time instead of linear time like the length check.

Re: `positive?` and `zero?` - yeah I haven't been wowed by them in practice either. The rules fire very often, but I haven't really seen them be worthwhile except in cases where they let you get rid of a lambda.